packages/starlight/__tests__/basics/format-path.test.ts

```diff
@@ -0,0 +1,105 @@
+			// with file extension and trailing slash
+			{ path: '/guides/components.html/', expected: '/guides/components.html' },
+		],
+	},
+	{
+		options: { format: 'file', trailingSlash: 'always' },
+		tests: [
+			// index page
+			{ path: '/', expected: '/index.html' },
+			// with trailing slash
+			{ path: '/reference/configuration/', expected: '/reference/configuration.html' },
+			// without trailing slash
+			{ path: '/api/v1/users', expected: '/api/v1/users.html' },
+			// with file extension
+			{ path: '/guides/components.html', expected: '/guides/components.html' },
+			// with file extension and trailing slash
+			{ path: '/guides/components.html/', expected: '/guides/components.html' },
+		],
+	},
+	{
+		options: { format: 'file', trailingSlash: 'never' },
+		tests: [
+			// index page
+			{ path: '/', expected: '/index.html' },
+			// with trailing slash
+			{ path: '/reference/configuration/', expected: '/reference/configuration.html' },
+			// without trailing slash
+			{ path: '/api/v1/users', expected: '/api/v1/users.html' },
+			// with file extension
+			{ path: '/guides/components.html', expected: '/guides/components.html' },
+			// with file extension and trailing slash
+			{ path: '/guides/components.html/', expected: '/guides/components.html' },
+		],
+	},
+	{
+		options: { format: 'directory', trailingSlash: 'always' },
+		tests: [
+			// index page
+			{ path: '/', expected: '/' },
+			// with trailing slash
+			{ path: '/reference/configuration/', expected: '/reference/configuration/' },
+			// without trailing slash
+			{ path: '/api/v1/users', expected: '/api/v1/users/' },
+			// with file extension
+			{ path: '/guides/components.html', expected: '/guides/components/' },
+			// with file extension and trailing slash
+			{ path: '/guides/components.html/', expected: '/guides/components/' },
+		],
+	},
+	{
+		options: { format: 'directory', trailingSlash: 'never' },
+		tests: [
+			// index page
+			{ path: '/', expected: '/' },
+			// with trailing slash
+			{ path: '/reference/configuration/', expected: '/reference/configuration' },
+			// without trailing slash
+			{ path: '/api/v1/users', expected: '/api/v1/users' },
+			// with file extension
+			{ path: '/guides/components.html', expected: '/guides/components' },
+			// with file extension and trailing slash
+			{ path: '/guides/components.html/', expected: '/guides/components' },
+		],
+	},
+	{
+		options: { format: 'directory', trailingSlash: 'ignore' },
+		tests: [
+			// index page
+			{ path: '/', expected: '/' },
+			// with trailing slash
+			{ path: '/reference/configuration/', expected: '/reference/configuration/' },
+			// without trailing slash
+			{ path: '/api/v1/users', expected: '/api/v1/users' },
+			// with file extension
+			{ path: '/guides/components.html', expected: '/guides/components' },
+			// with file extension and trailing slash
+			{ path: '/guides/components.html/', expected: '/guides/components' },
```

Comment:
🤔 I honestly don’t know… `/guides/components.html/` is already so totally badly formatted that it shouldn’t happen. I even debated if we should handle this case or only strip `.html` from paths without a trailing slash. Because on some hosts this would actually point to `/guides/components.html/index.html` 😅 

I think I’m happy with calling this one of those “bugs” we don’t have to fix until someone actually has a use case where this is an issue.